### ISSUE TITLE
[make] When using make with --warn-undefined-variables -R options getting a warning with correct makefile or no makefile

### ISSUE THREAD

**Issue by aandrejevas** [msys2/MINGW-packages#21146]
### Description / Steps to reproduce the issue

1. pacman -S mingw-w64-ucrt-x86_64-make or pacman -S make
2. mingw32-make --warn-undefined-variables -R or make --warn-undefined-variables -R
3. Warning is printed: "make: warning: undefined variable 'GNUMAKEFLAGS'"

The warning is not shown if GNUMAKEFLAGS environment variable is defined or it is defined in a makefile or by an option -E "GNUMAKEFLAGS:=". I think somewhere internal make code uses that variable but it is undefined because of the -R option so it shows a warning because of the --warn-undefined-variables option.

### Expected behavior

make would not show that warning.

### Actual behavior

![image](https://github.com/msys2/MINGW-packages/assets/53169542/ca78f634-c885-4323-9419-5cc901ad23e0)

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MSYS_NT-10.0-19045

### MINGW environments affected

- [ ] MINGW64
- [ ] MINGW32
- [X] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

**Comment by Biswa96** [comment]
Does this happen with a specific project?

**Comment by aandrejevas** [comment]
@Biswa96 No. This happens when not using a makefile (see screenshot in issue) and with any project whose makefile does not define GNUMAKEFLAGS variable.

**Comment by aandrejevas** [comment]
I found that this bug is already reported but it does not seem to be fixed:
https://lists.gnu.org/archive/html/bug-make/2022-11/msg00196.html
https://savannah.gnu.org/bugs/?63439
